app/models/user.rb

```diff
@@ -6,6 +6,8 @@ class User < ActiveRecord::Base
   has_many :reviews , -> { order('created_at DESC') } 
   has_many :following_friendships, class_name: 'Friendship', foreign_key: :follower_id
   
+  before_create :generate_token 
```

Comment:
Generating token at users creation is a huge security risk. Whoever would take possession or steal your database data, would access all accounts, because token works like a password - if you know the token, you can change PW and then there is no point in password encryption. You should create tokens at least users request and try to expire them after 1 hour. See Railscasts on how to expire.